public/res/js/slider.js

```diff
@@ -0,0 +1,28 @@
+let slideIndex = 1;
+showSlides(slideIndex);
+
+// Next/previous controls
+function plusSlides(n) {
+    showSlides(slideIndex += n);
+}
+
+// Thumbnail image controls
+function currentSlide(n) {
+    showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+    let i;
+    let slides = document.getElementsByClassName("mySlides");
```

Comment:
DOM elements should be cached on load of this JS, and should not be recalculated/traversed on each showslide call